raiden/api/rest.py

```diff
@@ -257,6 +258,16 @@ def open(self, partner_address, token_address, settle_timeout, balance=None):
             )
         except (InvalidAddress, InvalidSettleTimeout, SamePeerAddress) as e:
             return make_response(str(e), httplib.CONFLICT)
+        except ValueError as e:
+            if re.match("Token address 0x[0-9a-f]{40} does not contain code", str(e)):
+                return make_response(str(e), httplib.CONFLICT)
+            elif re.match("Manager for token 0x[0-9a-f]{40} does not exist", str(e)):
+                return make_response(str(e), httplib.CONFLICT)
+            raise
+        except Exception as e:
+            if re.match("^Duplicated channel$", str(e)):
```

Comment:
I'm not sure this works, as `str(e)` would return the whole exception description, including class name. You should use `e.message`. Also, doing a `re.match` against a simple string, with `^` and `$` is less efficient but the same as a simple `==` comparison, or you could prefer `in` operator. Anyway, for specific message/exception handling, comment above for own exception class, applies.